examples/motor-imagery/helpers/frequencybandselection_helpers.py

```diff
@@ -0,0 +1,226 @@
+
+
+def freq_selection_class_dis(raw, freq_band=[5., 35.], sub_band_width=4,
+                             sub_band_step=2, alpha=0.4,
+                             tmin=0.5, tmax=2.5,
+                             picks=None, event_id=None,
+                             cv=None, train_ind=None,
+                             method='cross_validation',
+                             return_class_dis=False, verbose=None):
+    r"""Select optimal frequency band based on class distinctiveness measure.
+
+    Optimal frequency band is selected by combining a filter bank with
+    a heuristic based on class distinctiveness on the manifold [1]_:
+
+    1. Filter training raw EEG data for each sub-band of a filter bank.
+    2. Estimate covariance matrices of filtered EEG for each sub-band.
+    3. Measure the class distinctiveness of each sub-band using the
+      classDis metric.
+    4. Find the optimal frequency band by starting from the sub-band
+      with the largest classDis and expanding the selected frequency band
+      as long as the classDis exceeds the threshold :math:`th`:
+
+    .. math::
+    th = max(classDis) - (max(classDis)−min(classDis)) × alpha
+
+
+    Parameters
+    ----------
+    raw : Raw object
+        An instance of Raw from MNE.
+    freq_band : list, default=[5., 35.]
+        Frequency band for band-pass filtering.
+    sub_band_width : float, default=4.
+        Frequency bandwidth of filter bank.
+    sub_band_step : float, default=2.
+        Step length of each filter bank.
+    alpha : float, default=0.4
+        Parameter to define an appropriate threshold for each individual.
+    tmin, tmax : float, default=0.5, 2.5
+        Start and end time of the epochs in seconds, relative to
+        the time-locked event.
+    picks : str | array_like | slice,  default=None
+        Channels to include. Slices and lists of integers will be
+        interpreted as channel indices.
+        If None (default), all channels will pick.
+    event_id : int | list of int | dict, default=None
+        Id of the events to consider.
+
+        - If dict, the keys can later be used to access associated
+          events.
+        - If int, a dict will be created with the id as string.
+        - If a list, all events with the IDs specified in the list
+          are used.
+        - If None, all events will be used and a dict is created
+          with string integer names corresponding to the event id integers.
+    cv : cross-validation generator, default=None
+        An instance of a cross validation iterator from sklearn.
+        If method is "cross_validation", cv needs to be defined.
+    train_ind : ndarray, shape (n_training_set,), default=None
+        index of training set.
+        If method is "train_test_split", train_ind needs to be defined.
+    method : str, default="cross_validation",
+        Method of data split either cross_validation or train_test_split.
```

Comment:
a train test split is nothing more than a ShuffleSplit with only 1 split. I would not make this a special case. You should get away with just one cv parameter I would say.

Reply:
I see! I did not know the link between train_test_split and ShuffleSplit. That's good to know!